src/Node/DirectoryInterface.php

```diff
@@ -28,6 +28,14 @@ public function createRecursive($mode = AdapterInterface::CREATION_MODE);
      */
     public function remove();
 
+    /**
+     * Rename the directory and return the new directory through a promise
+     *
+     * @param string $toDirectoryName
+     * @return PromiseInterface<FileInterface>
```

Comment:
Shouldn't this be `PromiseInterface<DirectoryInterface>`?

Reply:
Yes, sure! Fixed

Reply:
Awesome 👍